llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -9584,7 +9584,7 @@ static bool processLoopInVPlanNativePath(
     ProfileSummaryInfo *PSI, LoopVectorizeHints &Hints,
     LoopVectorizationRequirements &Requirements) {
 
-  if (isa<SCEVCouldNotCompute>(PSE.getBackedgeTakenCount())) {
+  if (isa<SCEVCouldNotCompute>(PSE.getSymbolicMaxBackedgeTakenCount())) {
```

Comment:
Does the change the behavior to allow vectorizing loops where getBackedgeTakenCount is SCEVCouldNotCompute? May need a test case and would be an unintended change?

Reply:
You're absolutely right. I was being too eager! I looked and there is no defence against loops with uncountable exits, since LoopVectorizationLegality doesn't filter those out in advance. I've reverted this change.

Reply:
Would it be possible to add a test case?

Reply:
Given my patch no longer changes this code are you asking if I can add a negative test for the native vplan case that proves we don't vectorise outer loops with an early exit? I can have a look, although I'm not sure that's even possible.

Reply:
Anyway, not sure if this is what you wanted or not but I added a test for an outer loop with an early exit. It didn't seem to be covered by any existing tests, although the legality code bails out too early to even attempt calculating a backedge taken count, due to the form of the conditional branch that is rejected by canVectorizeOuterLoop. I hacked the legality code and verified that we can get a symbolic max backedge take count for the outer loop, but not a normal one.

Reply:
>  can add a negative test for the native vplan case that proves we don't vectorise outer loops with an early exit?

yes that was what I had in mind to guard against the regression in the future, thanks!